app-editors/zed/zed-0.185.15.ebuild

```diff
@@ -0,0 +1,235 @@
+	[tree-sitter-md]='https://github.com/tree-sitter-grammars/tree-sitter-markdown;9a23c1a96c0513d8fc6520972beedd419a973539;tree-sitter-markdown-%commit%'
+	[tree-sitter-yaml]='https://github.com/zed-industries/tree-sitter-yaml;baff0b51c64ef6a1fb1f8390f3ad6015b83ec13a;tree-sitter-yaml-%commit%'
+	[webrtc-sys-build]='https://github.com/zed-industries/livekit-rust-sdks;80bb8f4c9112789f7c24cc98d8423010977806a6;livekit-rust-sdks-%commit%/webrtc-sys/build'
+	[webrtc-sys]='https://github.com/zed-industries/livekit-rust-sdks;80bb8f4c9112789f7c24cc98d8423010977806a6;livekit-rust-sdks-%commit%/webrtc-sys'
+	[xim-ctext]='https://github.com/XDeme1/xim-rs;d50d461764c2213655cd9cf65a0ea94c70d3c4fd;xim-rs-%commit%/xim-ctext'
+	[xim-parser]='https://github.com/XDeme1/xim-rs;d50d461764c2213655cd9cf65a0ea94c70d3c4fd;xim-rs-%commit%/xim-parser'
+	[xim]='https://github.com/XDeme1/xim-rs;d50d461764c2213655cd9cf65a0ea94c70d3c4fd;xim-rs-%commit%'
+)
+
+LLVM_COMPAT=( 19 )
+RUST_MIN_VER="1.86.0"
+RUST_NEEDS_LLVM=1
+WEBRTC_COMMIT="b99fd2c-6"
+
+inherit cargo check-reqs desktop flag-o-matic llvm-r1 toolchain-funcs xdg
+
+DESCRIPTION="The fast, collaborative code editor"
+HOMEPAGE="https://zed.dev https://github.com/zed-industries/zed"
+SRC_URI="
+	https://github.com/zed-industries/zed/archive/refs/tags/v${PV/_/-}.tar.gz -> ${P}.tar.gz
+	https://gitlab.com/api/v4/projects/35204985/packages/generic/${PN}/${PV}/${P}-crates.tar.xz
+	amd64? (
+		https://github.com/livekit/rust-sdks/releases/download/webrtc-${WEBRTC_COMMIT}/webrtc-linux-x64-release.zip ->
+			webrtc-${WEBRTC_COMMIT}-linux-x64-release.zip
+	)
+	arm64? (
+		https://github.com/livekit/rust-sdks/releases/download/webrtc-${WEBRTC_COMMIT}/webrtc-linux-arm64-release.zip ->
+			webrtc-${WEBRTC_COMMIT}-linux-arm64-release.zip
+	)
+	${CARGO_CRATE_URIS}"
+
+S="${WORKDIR}/${PN}-${PV/_/-}"
+LICENSE="GPL-3+"
+# Dependent crate licenses
+LICENSE+="
+	Apache-2.0 Apache-2.0-with-LLVM-exceptions BSD-2 BSD CC0-1.0 ISC
+	LGPL-3 MIT MIT-0 MPL-2.0 UoI-NCSA openssl Unicode-3.0 ZLIB
+"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64"
+IUSE="gles lto"
+CHECKREQS_DISK_BUILD="11G"
+CHECKREQS_MEMORY="16G"
+
+DEPEND="
+	app-arch/zstd:=
+	app-misc/jq
+	dev-db/sqlite:3
+	>=dev-libs/libgit2-1.9.0:=
+	dev-libs/mimalloc
+	dev-libs/openssl:0/3
+	dev-libs/protobuf
+	dev-libs/wayland
+	dev-libs/wayland-protocols
+	dev-util/wayland-scanner
+	dev-util/vulkan-tools
+	|| (
+		media-fonts/dejavu
+		media-fonts/cantarell
+		media-fonts/noto
+		media-fonts/ubuntu-font-family
+	)
+	media-libs/alsa-lib
+	media-libs/fontconfig
+	media-libs/vulkan-loader[X]
+	net-misc/curl
+	sys-libs/zlib
+	x11-libs/libX11
+	x11-libs/libxcb:=
+	x11-libs/libxkbcommon[X]
+"
+RDEPEND="${DEPEND}"
+BDEPEND="
+	app-arch/unzip
+	dev-build/cmake
+	dev-util/vulkan-headers
+	sys-devel/gettext
+	sys-devel/mold
+	$(llvm_gen_dep '
+		llvm-core/clang:${LLVM_SLOT}=
+		llvm-core/llvm:${LLVM_SLOT}=
+	')
+"
+
+QA_FLAGS_IGNORED="usr/bin/zedit"
+
+pkg_setup() {
+	if use lto; then
```

Comment:
We're trying to get rid of the use flag here. Since it doesn't pull any dependencies, you can rely on `tc-is-lto` function from the *toolchain-funcs.eclass*.

Reply:
I was discussing the same thing with Sam! :)

There's a nuance here that I'm not fully sure how to answer, which is that tc-is-lto checks to see if $CC supports lto, not RUSTC -- as we don't have RUSTFLAGS specced out for make.conf

Note also that cargo.eclass filters lto globally for C/C++ because rustc mixes compiler bytecode formats so it doesn't work. Although actually this should be fine here because it's run before the eclass code.

Reply:
Using `tc-is-lto` now